frontend/datetime.lua

```diff
@@ -307,4 +307,23 @@ function datetime.secondsToDateTime(seconds, twelve_hour_clock, use_locale)
     return message_text
 end
 
+--- Converts a date+time string to seconds
+---- @string "YYYY-MM-DD HH:MM:SS", time may be absent
+---- @treturn seconds
+function datetime.stringToSeconds(datetime_string)
+    if not datetime_string:match("%d+:%d+:%d+") then
+        datetime_string = datetime_string .. " 00:00:00"
+    end
+    local year, month, day, hour, min, sec = datetime_string:match("(%d+)-(%d+)-(%d+) (%d+):(%d+):(%d+)")
+    local dateTimeTable = {
```

Comment:
Our coding style advises thie kind of casing for methods/functions.
So, best to use underscore for these kind of variable name, ie. `datetime_table` or `datetime_t` or `d` or `t` (it is used just after, no real need to have an explicite name - you could even inline it inside os.time(...).